db_dtypes/__init__.py

```diff
@@ -263,7 +266,16 @@ def _datetime(
             year = int(match.group("year"))
             month = int(match.group("month"))
             day = int(match.group("day"))
-            return pandas.Timestamp(year=year, month=month, day=day).to_datetime64()
+
+            dateObj = pandas.Timestamp(
+                year=year,
+                month=month,
+                day=day,
+            )
+            if pandas.Timestamp.min < dateObj < pandas.Timestamp.max:
+                return dateObj.to_datetime64()
+            else:
+                raise OutOfBoundsDatetime("Out of bounds", scalar)
```

Comment:
To make coverage happy, the easiest thing is to exclude these lines and this branch:

```suggestion
            else:  # pragma: NO COVER
                # TODO(#166): Include these lines in coverage when pandas 2.0 is released.
                raise OutOfBoundsDatetime("Out of bounds", scalar)  # pragma: NO COVER
```